src/server/api/controllers/candidates.controller.js

```diff
@@ -43,8 +43,26 @@ const createCandidate = async (userId, boardId, newCandidate) => {
   return createNewCandidate;
 };
 
+const getCandidates = async (userId, boardId) => {
+  if (!Number.isInteger(Number(userId)) || !Number.isInteger(Number(boardId))) {
+    throw new InvalidIdError('Id should be an integer');
+  }
+  const candidates = await knex('candidates')
+    .join('boards', 'candidates.boardId', '=', 'boards.id')
```

Comment:
You don't need to join on the boards table, the `boardId` is already present on the other tables and you don't need other information from that table.